src/algebra/CommRing/adjunctions.lean

```diff
@@ -19,23 +21,27 @@ namespace CommRing
 
 local attribute [instance, priority 0] classical.prop_decidable
 
-noncomputable def polynomial_ring : Type u ⥤ CommRing.{u} :=
-{ obj := λ α, ⟨mv_polynomial α ℤ⟩,
-  map := λ α β f, ⟨rename f, by apply_instance⟩ }
+def CommRing.free (α : Type u) : CommRing.{u} := ⟨mv_polynomial α ℤ⟩
```

Comment:
I'm not very happy with this...

1. I think the full name is now `CommRing.CommRing.free`.
2. This feels a bit like mixing two naming conventions: `polynomial` vs `free`.

Would Lean accept something like `polynomial_ring.obj`? Or otherwise with `_` instead of `.`?

In hindsight I shouldn't have called the functor `polynomial_ring`. It should probably be `mv_polynomial_int` or something like that...

Reply:
I don't really want to have that definition at all, but the elaborator is getting confused about types if I inline it... 

How about we just call this functor `CommRing.free`? Then I'll name the current helper functor `free` as `free_obj`.

Reply:
How about this?